### PR TITLE
Feature/23 sequelize functions

### PR DESCRIPTION
**Pull request by JDziewonski98**
# Summary

Added magical sequelize functions to collection for @SiboYang 

## Test Plan
unit tests

## Related Issues

23

### CODE REVIEW

**Comment by JDziewonski98** [comment]
same stuff i already did for tour

File: backend/src/models/muralcollection.model.ts [hunk 1]
```diff
@@ -1,10 +1,33 @@
...
   public id!: number;
   public name!: string;
   public description!: string;
+
+  //pure virtual functions so that typescript can detect sequelize's magical runtime association functions at compile time.
+  public getMurals!: BelongsToManyGetAssociationsMixin<Mural>;
+  public addMural!: BelongsToManyAddAssociationMixin<Mural, number>;
+  public hasMural!: BelongsToManyHasAssociationMixin<Mural, number>;
+  public countMurals!: BelongsToManyCountAssociationsMixin;
+  public createMural!: BelongsToManyCreateAssociationMixin<Mural>;
```
Comment (tedspare) [inline backend/src/models/muralcollection.model.ts:24]: Is there a remove / `BelongsToManyRemoveAssociationMixin`?

File: backend/src/models/tests/muralcollection.test.ts [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+import { MuralCollection } from "../muralcollection.model";
+
+beforeAll(async () => {
+  await Mural.belongsTo(Borough, {
+    foreignKey: { allowNull: false, name: "boroughId" },
+  });
+  await Mural.belongsTo(Artist, {
+    foreignKey: { allowNull: false, name: "artistId" },
+  });
+  Mural.belongsToMany(MuralCollection, {
```
Comment (tedspare) [inline backend/src/models/tests/muralcollection.test.ts:14]: Are these two also async?

File: backend/src/models/tests/muralcollection.test.ts [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+  );
+
+  const collection = await MuralCollection.findOne({
+    where: {
+      name: "findme",
+    },
+  });
+  expect(collection).not.toEqual(null);
+  expect(collection!.description).toEqual("asdasdasd2");
+  expect(collection!.name).toEqual("findme");
```
Comment (tedspare) [inline backend/src/models/tests/muralcollection.test.ts:95]: lol

**Review by tedspare — APPROVED** [review]
LGTM

**Review by zhenglinxiao — APPROVED** [review]
Still don't quite get all of this but looks good

File: backend/src/models/tests/muralcollection.test.ts (hunk 2 above)
Comment (JDziewonski98) [reply backend/src/models/tests/muralcollection.test.ts:14]: good catch

File: backend/src/models/muralcollection.model.ts (hunk 1 above)
Comment (JDziewonski98) [reply backend/src/models/muralcollection.model.ts:24]: there indeed is, i'll add it

**Pull request merged**